File: A1_Caesar_Cipher.cpp
#include <iostream>
using namespace std;

string encrypt(string text, int s)
{
	string result = "";
	unsigned int i;
	for (i = 0; i < text.length(); i++) {
		if (isupper(text[i]))
			result += char(int(text[i] + s - 65) % 26 + 65);
		else
			result += char(int(text[i] + s - 97) % 26 + 97);
	}
	return result;
}

string decrypt(string text, int s)
{
	string result = "";
	unsigned int i;
	for (i = 0; i < text.length(); i++) {
		if (isupper(text[i]))
			result += char(int(text[i] - s - 65) % 26 + 65);
		else
			result += char(int(text[i] - s - 97) % 26 + 97);
	}
	return result;
}

int main()
{
	string text;
	cout << "Text : ";
    cin>>text;

	int s;
	cout << "\nShift: ";
    cin>>s;

	string cipher;
	cipher = encrypt(text, s);
	cout << "\nAfter Encrpytion\nCipher: " << cipher;

	cout << "\n\nAfter Decryption\nPlain Text: " << decrypt(cipher, s);
	return 0;
}

File: A1_Caesar_Cipher.py
# Caesar Cipher Technique
def encrypt(text,s):
	result = ""
	for i in range(len(text)):
		ch = text[i]
		if (ch.isupper()):
			result += chr((ord(ch) + s - 65) % 26 + 65) # Encrypt uppercase characters
		else:
			result += chr((ord(ch) + s - 97) % 26 + 97) # Encrypt lowercase characters
	return result

def decrypt(text,s):
	result = ""
	for i in range(len(text)):
		ch = text[i]
		if (ch.isupper()):
			result += chr((ord(ch) - s - 65) % 26 + 65) # Decrypt uppercase characters
		else:
			result += chr((ord(ch) - s - 97) % 26 + 97) # Decrypt lowercase characters
	return result


text = input("\nPlain text: ")
s = int(input("Shift: "))
cipher = encrypt(text,s)
print("\nEncrypted Cipher Text: " + cipher)
plain = decrypt(cipher,s)
print("Decrypted Text : " + plain + "\n")